feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/utils.c

```diff
@@ -197,22 +197,32 @@ int phy_from_path(char *_path, char *phy, unsigned int idx)
 
 int phy_find_hwmon(char *phy, char *hwmon)
 {
-	char tmp[PATH_MAX];
+	char tmp_wifi5[PATH_MAX];
+	char tmp_wifi6[PATH_MAX];
 	glob_t gl;
 
 	*hwmon = '\0';
-	snprintf(tmp, sizeof(tmp), "/sys/class/ieee80211/%s/device/hwmon/*", phy);
-	if (glob(tmp, GLOB_NOSORT | GLOB_MARK, NULL, &gl))
-                return -1;
+	snprintf(tmp_wifi5, sizeof(tmp_wifi5), "/sys/class/ieee80211/%s/device/hwmon/*", phy);
+	snprintf(tmp_wifi6, sizeof(tmp_wifi6), "/sys/class/ieee80211/%s/cooling_device/subsystem/thermal_zone0/", phy);
+	if (!glob(tmp_wifi5, GLOB_NOSORT | GLOB_MARK, NULL, &gl)) {
+		if (gl.gl_pathc) {
+			strcpy(hwmon, gl.gl_pathv[0]);
+			strncat(hwmon, "temp1_input", PATH_MAX);
+			globfree(&gl);
+			return 0;
+		}
+	}
+	else if(glob(tmp_wifi6, GLOB_NOSORT | GLOB_MARK, NULL, &gl))
```

Comment:
Using glob() here and !glob() above at line 207 is bad for readability.

Really you have one operation made on two data sets:   
"/sys/class/ieee80211/%s/device/hwmon/*" with "temp1_input", and"/sys/class/ieee80211/%s/cooling_device/subsystem/thermal_zone0/" with "temp".

You could create a helper function that takes these two parameters as inputs, and &temp for the result, and that returns success/fail.  Then the code would be very easy to understand and you'd avoid the duplication and you'd have a helper function that could be used for other information (say CPU utilization maybe).

Reply:
sure, done